File: 560. Subarray Sum Equals K/main.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        vector<int> dp(nums.size()+1,0);
        for(int i=0;i<nums.size();i++){
            dp[i+1]=nums[i]+dp[i];
        }
        int c=0;
        for(int i=0;i<dp.size()-1;i++){
            for(int j=i+1;j<dp.size();j++){
                if(dp[j]-dp[i]==k)
                    c++;
            }
        }
        return c;
    }
};

File: 560. Subarray Sum Equals K/main.py
import collections
class Solution:
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        s=0
        c=0
        
        d=collections.defaultdict(int)
        d[0]=1
        for num in nums:
            s+=num
            if s-k in d.keys():
                c=c+d[s-k]
            d[s]+=1
            
        return c
